index.d.ts

```diff
@@ -1,4 +1,31 @@
 import { FastifyPluginCallback } from 'fastify'
+import {
+  ContextConfigDefault,
+  RawReplyDefaultExpression,
+  RawRequestDefaultExpression,
+  RawServerBase,
+  RawServerDefault,
+} from 'fastify/types/utils'
+import { FastifyInstance } from 'fastify/types/instance'
+import { FastifyRequest } from 'fastify/types/request'
+import { FastifyReply } from 'fastify/types/reply'
+import { RouteGenericInterface } from 'fastify/types/route'
+
+declare module 'fastify' {
+  interface RouteHandlerMethod<
+    RawServer extends RawServerBase = RawServerDefault,
+    RawRequest extends RawRequestDefaultExpression<RawServer> = RawRequestDefaultExpression<RawServer>,
+    RawReply extends RawReplyDefaultExpression<RawServer> = RawReplyDefaultExpression<RawServer>,
+    RouteGeneric extends RouteGenericInterface = RouteGenericInterface,
+    ContextConfig = ContextConfigDefault
+  > {
+    (
+      this: FastifyInstance<RawServer, RawRequest, RawReply>,
+      request: FastifyRequest<RouteGeneric, RawServer, RawRequest>,
+      reply: FastifyReply<RawServer, RawRequest, RawReply, RouteGeneric, ContextConfig>
+    ): void | Promise<RouteGeneric['Reply'] | void> | Record<string, any> | (() => any)
```

Comment:
Technically this works to get rid of the type error, but by using `any` we are losing the type-safety of `RouteGeneric`. It would be better if we could just extend this to cover the expected funkify return types. Something like 

```ts
type FunkifyReply<T> = {right: T} | {left: unknown} | (() => FunkifyReply<T>) // .. any other return types
```
And then we could do:
```suggestion
    ): void | Promise<RouteGeneric['Reply'] | void> | FunkifyReply<RouteGeneric['Reply']>
```

That allows users to do:

```ts
// this is allowed
fastify.get("/", (req, reply) => {
	return { right: { id: 1 } };
});
// ...but this gives a (correct) type error
fastify.get('/', (req: FastifyRequest, reply: FastifyReply) => {
  return { rght: { id: 1 } }
})

```